rcl/test/rcl/test_client.cpp

```diff
@@ -277,3 +278,25 @@ TEST_F(TestClientFixture, test_client_bad_arguments) {
       &client, &client_request, &sequence_number)) << rcl_get_error_string().str;
   EXPECT_EQ(24, sequence_number);
 }
+
+TEST_F(TestClientFixture, test_client_init_fini_maybe_fail)
+{
+  const rosidl_service_type_support_t * ts = ROSIDL_GET_SRV_TYPE_SUPPORT(
+    test_msgs, srv, BasicTypes);
+  constexpr char topic_name[] = "chatter";
+  rcl_client_options_t default_client_options = rcl_client_get_default_options();
+
+  RCUTILS_FAULT_INJECTION_TEST(
+  {
+    rcl_client_t client = rcl_get_zero_initialized_client();
+    rcl_ret_t ret = rcl_client_init(
+      &client, this->node_ptr, ts, topic_name, &default_client_options);
+    if (RCL_RET_OK == ret) {
+      EXPECT_TRUE(rcl_client_is_valid(&client));
+      ret = rcl_client_fini(&client, this->node_ptr);
```

Comment:
@brawner why not:
```suggestion
      EXPECT_EQ(RCL_RET_OK, rcl_client_fini(&client, this->node_ptr));
```
?

Reply:
What happens in this scenario if `init` doesn't return error but `fini` does?

Reply:
While there are few functions in `_fini` that can fail it still is possible with fault injection in some cases. I added a check and a second attempt to finalize with your suggested `EXPECT_EQ` check on the second one.